spotify_player/src/cli/client.rs

```diff
@@ -350,3 +365,453 @@ async fn handle_playback_request(
+    for track in from_tracks {
+        let song = PlaylistData {
+            id: track.id.to_owned(),
+            name: track.name,
+        };
+
+        from_ids.push(song);
+        hasher.write(track.id.id().as_bytes());
+    }
+    let hash = hasher.finish().to_string();
+
+    // Get import dir/file
+    let conf_dir = get_config_folder_path()?;
+    let imports_dir = conf_dir.join("imports");
+    let to_dir = imports_dir.join(import_to.id());
+    let from_file = to_dir.join(import_from.id());
+
+    if !to_dir.exists() {
+        create_dir_all(to_dir)?;
+    }
+
+    // If this import has not happened before
+    if !from_file.exists() {
+        let num_songs = from_ids.len();
+
+        let mut file = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(&from_file)?;
+
+        // Write hash and new line to top of file
+        writeln!(file, "{}", hash)?;
+        writeln!(file)?;
+
+        let mut track_buff = Vec::new();
+        for track in from_ids {
+            writeln!(file, "{}:{}", track.id.id(), track.name)?;
+
+            if !to_hash.contains(&track.id) {
+                track_buff.push(PlayableId::Track(track.id));
+
+                if track_buff.len() > 90 {
+                    client
+                        .spotify
+                        .playlist_add_items(import_to.to_owned(), track_buff, None)
+                        .await?;
+                    track_buff = Vec::new();
+                }
+            }
+        }
+
+        if !track_buff.is_empty() {
+            client
+                .spotify
+                .playlist_add_items(import_to.to_owned(), track_buff, None)
+                .await?;
+        }
+
+        Ok(format!(
+            "Successfully imported '{}' into '{}'.\n{} songs were added.",
+            from.name, to.name, num_songs
+        ))
+    } else {
+        // If the import has happened before, read from hash to check from changes
+        // If the hashes don't match reimport
+
+        let file = OpenOptions::new().read(true).open(&from_file)?;
+        let reader = BufReader::new(file);
+        let mut iter = reader.lines();
+
+        let old_hash = iter.next().unwrap()?;
+
+        // Consume empty line
+        iter.next();
+
+        // If old hash doesn't match,
+        // need to do import
+        if !old_hash.eq(&hash) {
+            let mut old_ids = HashSet::new();
+            let mut new_ids = HashSet::new();
+
+            // Write playlists new info to file
+            remove_file(&from_file)?; // Writing wasn't overwriting
+            let mut file = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .open(&from_file)?;
+
+            writeln!(file, "{}", hash)?;
+            writeln!(file)?;
+
+            for track in from_ids {
+                let id = track.id.to_owned();
+
+                writeln!(file, "{}:{}", id.id(), track.name)?;
```

Comment:
If you update the file cache by just storing the latest playlist's data. I don't think it helps with the "don't add deleted tracks" use case you mentioned in https://github.com/aome510/spotify-player/pull/222/files#r1277490271.

For example, let's say you initially import tracks "A B" from a playlist and then delete "B". File cache: "A B"

 You call a second import with tracks "A B C". File cache: "A B C", only "C" is imported.

You call a third import with tracks "A B C D". File cache: "A B C D", "B D" are imported.

This means the "B" deleted before is added back.